File: internal/payments/payment.go
package payments

import "time"

type Payment struct {
	Amount        float64
	CorrelationId string
	RequestedAt   time.Time
	Processor     ProcessorType
}

func NewPayment(amountCents float64, correlationId string, processor ProcessorType, requestedAt time.Time) *Payment {
	return &Payment{
		Amount:        amountCents,
		CorrelationId: correlationId,
		RequestedAt:   requestedAt,
		Processor:     processor,
	}
}

File: worker/src/payment.rs
use crate::processor_type::ProcessorType;
use rust_decimal::Decimal;
use time::OffsetDateTime;

#[derive(Debug, Clone)]
pub struct Payment {
    pub amount: Decimal,
    pub correlation_id: uuid::Uuid,
    pub requested_at: OffsetDateTime,
    pub processor: ProcessorType,
}

impl Payment {
    pub fn new(
        amount: Decimal,
        correlation_id: uuid::Uuid,
        processor: ProcessorType,
        now: OffsetDateTime,
    ) -> Self {
        Self {
            amount,
            correlation_id,
            processor,
            requested_at: now,
        }
    }
}
